zerver/lib/upload.py

```diff
@@ -149,6 +149,11 @@ def get_file_info(request, user_file):
     # type: (HttpRequest, File) -> Tuple[text_type, Optional[text_type]]
 
     uploaded_file_name = user_file.name
+    if not isinstance(user_file.name, str):
+        # This check makes sure that the type of the file name is not
+        # Python 2 Unicode.
+        uploaded_file_name = uploaded_file_name.encode('ascii')
```

Comment:
I think it probably makes sense to do this in the view, since the way the filename we receive is encoded will be a property of how we got the filename, and so I'm not sure this bug would apply to potential future other users of this library.